packages/flutter/lib/src/widgets/animated_switcher.dart

```diff
@@ -218,18 +218,29 @@ class AnimatedSwitcher extends StatefulWidget {
 
   /// A function that wraps a new [child] with an animation that transitions
   /// the [child] in when the animation runs in the forward direction and out
-  /// when the animation runs in the reverse direction. This is only called
-  /// when a new [child] is set (not for each build), or when a new
-  /// [transitionBuilder] is set. If a new [transitionBuilder] is set, then
-  /// the transition is rebuilt for the current child and all previous children
-  /// using the new [transitionBuilder]. The function must not return null.
+  /// when the animation runs in the reverse direction.
+  ///
+  /// This is only called when a new [child] is set (not for each build), [child]
```

Comment:
It's not called for any animation status change, is it? Should we maybe say "This is only called when a new [child] is set (not for each build), when [child] starts to transition out, or when a new [transitionBuilder] is set.